js/src/data/resolvers.js

```diff
@@ -746,16 +747,19 @@ export function* getPriceBenchmarkSuggestions( args ) {
 /**
  * Resolver for getting the Ads recommendations.
  */
-export function* getAdsRecommendations( types ) {
+export function* getAdsRecommendations( types, campaign_id = null ) {
 	try {
 		const response = yield apiFetch( {
 			path: addQueryArgs( `${ API_NAMESPACE }/ads/recommendations`, {
 				types,
+				campaign_id,
 			} ),
 		} );
 
-		const typesKey = arrayToUnderscoreKey( types );
-		yield receiveAdsRecommendations( response, typesKey );
+		const key = campaign_id ? [ campaign_id, ...types ] : types;
```

Comment:
@ankitrox We can do this as well:
```suggestion
		const key = [campaign_id, ...types].filter(Boolean);
```
which will remove any falsy values from the array.